src/methods/has.js

```diff
@@ -61,7 +61,6 @@ export function HasName(realm: Realm, ast: BabelNodeExpression): boolean {
 // ECMA262 7.3.10
 export function HasProperty(realm: Realm, O: ObjectValue | AbstractObjectValue, P: PropertyKeyValue): boolean {
   // 1. Assert: Type(O) is Object.
-  invariant(O instanceof ObjectValue || O instanceof AbstractObjectValue, "expected object");
 
```

Comment:
Writing an invariant that is obviously true, just because we don't trust the type checker deeply offends me.